### ISSUE TITLE
Console window visible in windows target

### ISSUE THREAD

**Issue by shashaanktulsyan** [gluonhq/client-maven-plugin#176]
Thank you implementing support for windows. 
The fast startup speed of native build is clearly felt. 
The build process was fairly simple and quick. 
The timeline in which this feature has been delivered surpasses expectation. 
To sum up, it was literally a thankless hard-work which the Gluon team has been doing and sharing the fruits of the hard work to the entire java community for free. And this I mean not just for windows support but also for mobile, android, ios everything put together.

Coming to the subject of this ticket, currently how the app is building seems to be as if it is executed using ``java.exe``. In windows we have this ``java.exe`` and ``javaw.exe`` If we run in java.exe we see both the command window and the UI, whereas ``javaw,exe`` suppress the console window.
This is fine for debugging but how do we stop the console window from showing?
This needs to be fixed, or if there is flag which needs to be set to enable/disable this behavior, that flag needs to be documented, and probably highlighted in bold or something to be easily noticeable in the docs page. 

Also I am sure soon other features like setting the icon of the executable will be added in future (if not already there). Even while it is experimental it should be clearly documented otherwise we might not come to know this without asking. 
Finally I think the client:package feature is pending implementation, which is ok, but I am just mentioning it because I don't know if it is not implemented yet, or it just didn't work for me. 
Probably a single status page (or just a human written page mentioning the status) of what is implemented and what is pending will help. If such a page is already there, it should be linked in the gluon docs. Sure there are tickets which might help in assessing what is done what is pending, but it is split across a projects like substrate and client-plugin etc, it wouldn't be easy to know. 

![image](https://user-images.githubusercontent.com/6197269/82390594-be623f80-9a5c-11ea-8491-9db1e4388d23.png)

Apart from what is mentioned in the guide https://docs.gluonhq.com/client/ which says

```
During the installation process, make sure that you select at least the following individual components:
- Choose the English Language Pack
- C++/CLI support for v142 build tools (14.25 or later)
- MSVC v142 - VS 2019 C++ x64/x86 build tools (v14.25 or later)
- Windows Universal CRT SDK
- Windows 10 SDK (10.0.19041.0 or later)
```

I had to also install **Build Tools for Visual Studios 2019** because first it said ``cannot find cl`` and then when I added this to the path, it said ``cannot find stdio.h`` 
I am documenting this so that others don't face the same problem. Probably the documentation could be updated to explain how to fix the cl not found and stdio.h not found error message.

Thank you

**Comment by tsatatwer** [comment]
> I had to also install Build Tools for Visual Studios 2019 because first it said cannot find cl and then when I added this to the path, it said cannot find stdio.h


I confirm that ......I faced the same issue and solved in a complex way .


And I added my voice to @shashaanktulsyan :

> Thank you implementing support for windows.
The fast startup speed of native build is clearly felt.
The build process was fairly simple and quick.
The timeline in which this feature has been delivered surpasses expectation.
To sum up, it was literally a thankless hard-work which the Gluon team has been doing and sharing the fruits of the hard work to the entire java community for free. And this I mean not just for windows support but also for mobile, android, ios everything put together.

Thanks for your great efforts you added a valuable work for Javafx and its developers .

**Comment by tiainen** [comment]
Tracked by https://github.com/gluonhq/substrate/issues/638

**Comment by shashaanktulsyan** [comment]
> Tracked by [gluonhq/substrate#638](https://github.com/gluonhq/substrate/issues/638)

Thank you so much. Should I close the ticket? This comes out in the next release right? Would you need me to test it?

**Comment by tiainen** [comment]
Yes, please test to see if it is indeed resolved.

**Comment by shashaanktulsyan** [comment]
@tiainen tried changing to client plugin version 0.1.26, it builds but the console window is still there. I tried to build on a 0.1.27-SNAPSHOT in case there was one, no success. Which version should I try? In the meantime i retrying with a clean and build command.

**Comment by tiainen** [comment]
You do indeed need to use the snapshot version `0.1.27-SNAPSHOT`. To be able to test it, you have to add the oss sonatype snapshots repository to your `pom.xml`:

    <repositories>
        <repository>
            <id>oss-sonatype-snapshots</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
       </repository>
    </repositories>

**Comment by shashaanktulsyan** [comment]
> <pluginRepositories>
>     <pluginRepository>
>         <id>oss-sonatype-snapshots</id>
>         <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
>    </pluginRepository>
> </pluginRepositories>

Yes it worked, I don't see console window anymore.
How do we set icons for the executable etc? 
No hurry, but kindly document once implemented in case not implemented, and if implemented already kindly share how it is done. 

Thank you.

**Comment by tiainen** [comment]
Fixed with version 0.1.27.

Please create a new issue for setting icons etc.
